src/org/zaproxy/zap/extension/scripts/ConsolePanel.java

```diff
@@ -104,6 +124,140 @@ private void initialize() {
 		panelContent.add(splitPane, LayoutHelper.getGBC(0, 1, 1, 1.0D, 1.0D));
 	}
 	
+	
+	private Method getScriptHasChangedOnDiskMethod() {
+		if (after2_7_0 == null) {
+			// not tried yet
+			try {
+				scriptHasChangedOnDiskMethod = ScriptWrapper.class.getMethod("hasChangedOnDisk");
+				after2_7_0 = Boolean.TRUE;
+			} catch (Exception e) {
+				after2_7_0 = Boolean.FALSE;
+			}
+		}
+		return scriptHasChangedOnDiskMethod;
+	}
+	
+	
+	private boolean isAfter2_7_0 () {
+		if (after2_7_0 == null) {
+			// Sets after2_7_0 as a side effect
+			getScriptHasChangedOnDiskMethod();
+		}
+		return after2_7_0;
+	}
+	
+	private boolean isScriptUpdatedOnDisk() {
+		if (script != null) {
+			// Check to see if its also changed on disk
+			try {
+				Object result = getScriptHasChangedOnDiskMethod().invoke(script);
+				if (result instanceof Boolean) {
+					return (Boolean)result;
+				} else {
+					LOG.error("ScriptWrapper.isChanged() unexpected return " + result + " " + result.getClass().getCanonicalName());
+				}
+			} catch (Exception e) {
+				LOG.error(e.getMessage(), e);
+			}
+		}
+	return false;
+	}
+
+	private void startPollingForChanges() {
+		changesPollingThread = new Thread() {
+			@Override
+			public void run() {
+				this.setName(THREAD_NAME);
+				pollForChanges = true;
+				while (pollForChanges) {
+					if (dialogRunnable == null && isScriptUpdatedOnDisk() ) {
+						
+						if (script.isChanged()) {

```

Comment:
It could use the same `Runnable` for both (use a variable for the message or use ternary operator in the invocation?).